tests/app/v2/notifications/test_post_notifications.py

```diff
@@ -362,6 +362,45 @@ def test_post_email_notification_returns_201(
     assert mock_deliver_email.called
 
 
+@pytest.mark.parametrize("reference", [None, "reference_from_client"])
+def test_post_email_notification_with_reply_to_returns_201(
+        client, sample_email_template_with_reply_to, mock_deliver_email, reference
+):
+    data = {
+        "email_address": sample_email_template_with_reply_to.service.users[0].email_address,
+        "template_id": sample_email_template_with_reply_to.id,
+        "personalisation": {"name": "Bob"},
+        "billing_code": "TESTCODE"
+    }
+    if reference:
```

Comment:
`if reference is not None:`

This is another instance of treating None as a boolean, which I assume you copied from the function test_post_email_notification_returns_201.  There seem to be many of them in the code, and we should fix them as we find them.